src/Ractive/prototype/set.js

```diff
@@ -10,9 +10,9 @@ export default function Ractive$set ( keypath, value ) {
 	if ( isObject( keypath ) ) {
 		const map = keypath;
 
-		for ( keypath in map ) {
-			if ( map.hasOwnProperty( keypath) ) {
-				set( this, keypath, map[ keypath ] );
+		for ( const k in map ) {
+			if ( map.hasOwnProperty( k) ) {
+				set( this, k, map[k] );
 			}
```

Comment:
Maybe there's some value in replacing `forEach` at some point then, particularly in hot code paths where speed > readability. Though catching other places where functions are being deoptimised is probably lower hanging fruit